### ISSUE TITLE
add custom fields/attribute to issues

### ISSUE THREAD

**Issue by andrewchenshx** [go-gitea/gitea#18210]
### Feature Description

I think it would be great if custom attributes/fields could be added to issues.
Taiga has this feature while it's too heavyweight compared to Gitea.
thanks :)

**Comment by andyculmer** [comment]
I agree that this would be a valuable feature. I would certainly be keen to migrate more projects to Gitea if custom fields were supported.

I note that this is a duplicate of #8665 and #20466, which were closed as not planned on the 24th July 2022.

The fact that 3 issues have been raised suggests there is some interest in this feature.

**Comment by andyculmer** [comment]
GitHub has had this feature for 2 years now, but I notice there's been no further comment on this proposal.

The issue tracker seems to be one area where Gitea doesn't quite match GitHub's feature set. I appreciate that Gitea does not have the development resources that GitHub has, so it would be great to understand whether this is a gap that's likely to be closed in the next year or so, or whether there's perhaps a strategic reason that this isn't being looked at. For example, is the strategy to defer to an external issue tracker for those that need this level of functionality and keep development effort focused on those areas that can't be so easily offloaded?

Any insights would be genuinely appreciated and would help with our longer term planning for tool selection. We may also be in a position to develop and contribute patches for some of this functionality, but I'm concerned that we could be paddling up stream.

**Comment by lunny** [comment]
I think yes, we have far less resources than Github.

Can you link some documentation or design about how Github support custom fields? I never used this feature on Github.

**Comment by andyculmer** [comment]
Thanks for your reply @lunny 

GitHub adds two more types of _View_ in addition to Gitea's Kanban board. These are _Roadmap_ and _Table_. In the _Table_ view, you can not only customize the columns in the table to add the standard issue fields, but you can also create new fields.

The documentation is as follows:
 * [Custom fields](https://docs.github.com/en/issues/planning-and-tracking-with-projects/understanding-fields/about-text-and-number-fields)
 * [The different view layouts](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/changing-the-layout-of-a-view)
 * [The tables view](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/customizing-the-table-layout)

**Comment by lunny** [comment]
I have a PR #30205, maybe it's related.

**Comment by andyculmer** [comment]
Some great features there, but looks mainly focused on labels. I didn't see any mention of custom fields in the list of resolved issues.

**Comment by KevinKrueger** [comment]
Table View with custom fields....... game changer! Excel good bye!

**Comment by Rehab42** [comment]
I just started with Gitea coming from GitHub and used the custom fields for story points to prioritize what I thought would take a while and or adding a priority field to then order the kanban view. Would really like a grid view as well.

**Comment by m384wwn** [comment]
This would be huge. We want to be able to write customer-facing release or patch notes, which would be a good candidate for a custom field. This way, they can easily be compiled at release time among other things.
